### ISSUE TITLE
S1643 should not apply when strings are not aggregated

### ISSUE THREAD

**Issue by m-gallesio** [SonarSource/sonar-dotnet#5521]
*Sonar version: Version 9.3 (build 51899)*

I do not think rule S1643 (strings should not be concatenated using '+' in a loop) should be applied when the loop is not used to aggregate strings.

Consider this example:

```
class MyObject
{
    public string Name { get; set; }
    public bool Enabled { get; set; }
}

void MarkDisabled(IList<MyObject> objects)
{
    foreach (var obj in objects)
    {
        if (!obj.Enabled)
            obj.Name += " - DISABLED";
    }
}
```

This alters a string field of a specific object and does not aggregate the results.

**Comment by m-gallesio** [comment]
This still applies to 9.4.

**Comment by costin-zaharia-sonarsource** [comment]
Thanks @m-gallesio for reporting this. I can confirm the false positive.

**Comment by m-gallesio** [comment]
Bump, this still applies to 9.9.

**Comment by m-gallesio** [comment]
This still applies to 10.1.

**Comment by martin-strecker-sonarsource** [comment]
Root cause: The string concatenation is happening on the loop variable. We need to exclude such cases.
